Apps/AccountDataAccess/src/Patient/PatientRepository.cs

```diff
@@ -80,55 +89,103 @@ public async Task<PatientQueryResult> Query(PatientQuery query, CancellationToke
             };
         }
 
-        private async Task<ApiResult<PatientModel>> GetDemographicsAsync(OidType type, string identifier, PatientIdentifierType identifierType, bool disableIdValidation = false)
+        private static bool ShouldCheckCache(PatientDetailSource source)
         {
-            ApiResult<PatientModel> apiResult = new()
+            return source switch
             {
-                ResourcePayload = this.GetFromCache(identifier, identifierType),
+                PatientDetailSource.AllCache => true,
+                PatientDetailSource.EmpiCache => true,
+                PatientDetailSource.PhsaCache => true,
+                _ => false,
```

Comment:
Just a share after curiosity with overlapping conditions, in C# 9, this can be rewritten as
```CSharp
return source switch
            {
                PatientDetailSource.AllCache or
                PatientDetailSource.EmpiCache or
                PatientDetailSource.PhsaCache => true,
                _ => false,
             };
```

Or in older versions

```CSharp
return source switch
            {
                var x when x.In(PatientDetailSource.AllCache, 
                PatientDetailSource.EmpiCache,
                PatientDetailSource.PhsaCache) => true,
                _ => false,
             };
```